File: app/src/main/java/testpreference/com/testcalendar/MainActivity.java
package testpreference.com.testcalendar;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.Date;
import java.util.Enumeration;
import java.util.Hashtable;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";
    private Spinner calendarIdSpinner;
    private Hashtable<String, String> calendarIdTable;
    private Button newEventButton;
    private Button deleteEvent;
    private Button EditEvent;
    private Button cancelEvent;
    private Button eventExist;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        calendarIdSpinner = (Spinner) findViewById(R.id.calendarid_spinner);
        newEventButton = (Button) findViewById(R.id.newevent_button);
        deleteEvent = (Button) findViewById(R.id.deleteEvent);
        EditEvent = (Button) findViewById(R.id.editEvent);
        cancelEvent = (Button) findViewById(R.id.cancelEvent);
        eventExist = (Button) findViewById(R.id.eventExist);
        newEventButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (CalendarHelper.haveCalendarReadWritePermissions(MainActivity.this)) {
                    addNewEvent();
                } else {
                    CalendarHelper.requestCalendarReadWritePermission(MainActivity.this);
                }
            }
        });

        deleteEvent.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                CalendarHelper.deleteEvent(MainActivity.this, 6589);
            }
        });
        eventExist.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                CalendarHelper.checkEventsInCal(MainActivity.this, 6589);

            }
        });

        EditEvent.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                CalendarHelper.editEvent(MainActivity.this, 6589);
            }
        });
        cancelEvent.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                CalendarHelper.cancelEvent(MainActivity.this, 6589);
            }
        });

        calendarIdSpinner.setOnItemSelectedListener(new CustomOnItemSelectedListener());

        System.out.println("primary calendar id is " + CalendarHelper.getCalendarId(this) + "");


    }

    @Override
    protected void onResume() {
        super.onResume();
        if (calendarIdTable != null) {
            calendarIdTable.clear();
        }
        if (CalendarHelper.haveCalendarReadWritePermissions(this)) {
            //Load calendars
            calendarIdTable = CalendarHelper.listCalendarId(this);

            updateCalendarIdSpinner();

        }

    }

    private void updateCalendarIdSpinner() {
        if (calendarIdTable == null) {
            return;
        }

        List<String> list = new ArrayList<String>();

        Enumeration e = calendarIdTable.keys();
        while (e.hasMoreElements()) {
            String key = (String) e.nextElement();
            list.add(key);
        }

        ArrayAdapter<String> dataAdapter = new ArrayAdapter<String>(this,
                android.R.layout.simple_spinner_item, list);
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        calendarIdSpinner.setAdapter(dataAdapter);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {

        if (requestCode == CalendarHelper.CALENDARHELPER_PERMISSION_REQUEST_CODE) {
            if (CalendarHelper.haveCalendarReadWritePermissions(this)) {
                Toast.makeText(this, (String) "Have Calendar Read/Write Permission.",
                        Toast.LENGTH_LONG).show();
                updateCalendarIdSpinner();
            }

        }

        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }

    private void addNewEvent() {
        if (calendarIdTable == null) {
            Toast.makeText(this, (String) "No calendars found. Please ensure at least one google account has been added.",
                    Toast.LENGTH_LONG).show();
            //Load calendars
            calendarIdTable = CalendarHelper.listCalendarId(this);

            updateCalendarIdSpinner();

            return;
        }

        final long oneHour = 1000 * 60 * 60;
        final long tenMinutes = 1000 * 60 * 5;

        long oneHourFromNow = (new Date()).getTime() + oneHour;
        long tenMinutesFromNow = (new Date()).getTime() + tenMinutes;

        CalendarHelper.MakeNewCalendarEntry(6589,this, "Test", "Add event", "Somewhere", tenMinutesFromNow, tenMinutesFromNow + oneHour, false, true, CalendarHelper.getCalendarId(this), 3);

    }


}

File: app/src/main/java/testpreference/com/testcalendar/MainActivity.kt
package testpreference.com.testcalendar

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
    private var calendarIdSpinner: Spinner? = null
    private var calendarIdTable: Hashtable<String, String>? = null
    private var newEventButton: Button? = null
    private var deleteEvent: Button? = null
    private var EditEvent: Button? = null
    private var cancelEvent: Button? = null
    private var eventExist: Button? = null
    private var addingInLoop: Button?=null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        calendarIdSpinner = findViewById(R.id.calendarid_spinner) as Spinner?
        newEventButton = findViewById(R.id.newevent_button) as Button?
        deleteEvent = findViewById(R.id.deleteEvent) as Button?
        EditEvent = findViewById(R.id.editEvent) as Button?
        cancelEvent = findViewById(R.id.cancelEvent) as Button?
        eventExist = findViewById(R.id.eventExist) as Button?
        addingInLoop = findViewById(R.id.addingInLoop) as Button?
        newEventButton?.setOnClickListener {
            if (CalendarHelper.haveCalendarReadWritePermissions(this@MainActivity)) {
                addNewEvent()
            } else {
                CalendarHelper.requestCalendarReadWritePermission(this@MainActivity)
            }
        }
        addingInLoop?.setOnClickListener {
            for (i in 0..3){
                val oneHour = (1000 * 60 * 60).toLong()
                val tenMinutes = (1000 * 60 * 5).toLong()

                val oneHourFromNow = Date().time + oneHour
                val tenMinutesFromNow = Date().time + tenMinutes

                val calendar = Calendar.getInstance()

                calendar.add(Calendar.DAY_OF_YEAR, 1+i)
                val tomorrow = calendar.time


                CalendarHelper.MakeNewCalendarEntry(6590+i, this, "Test", "Add event", "Somewhere", tomorrow.time + tenMinutes,tomorrow.time + oneHour, false, true, CalendarHelper.getCalendarId(this), 3)

            }
        }
        deleteEvent?.setOnClickListener {
            CalendarHelper.deleteEvent(this@MainActivity, 6589)
        }
        eventExist?.setOnClickListener {
            CalendarHelper.checkEventsInCal(this@MainActivity, 6589)
        }

        EditEvent?.setOnClickListener {
            val oneHour = (1000 * 60 * 60).toLong()
            val tenMinutes = (1000 * 60 * 10).toLong()

            val calendar = Calendar.getInstance()

            calendar.add(Calendar.DAY_OF_YEAR, 1)
            val tomorrow = calendar.time

            CalendarHelper.editEvent(this@MainActivity, 6589,tomorrow.time + tenMinutes,tomorrow.time + oneHour)
        }
        cancelEvent?.setOnClickListener {
            CalendarHelper.cancelEvent(this@MainActivity, 6589)
        }

        calendarIdSpinner?.onItemSelectedListener = CustomOnItemSelectedListener()

    }

    override fun onResume() {
        super.onResume()
        if (calendarIdTable != null) {
            calendarIdTable!!.clear()
        }
        if (CalendarHelper.haveCalendarReadWritePermissions(this)) {
            //Load calendars
            calendarIdTable = CalendarHelper.listCalendarId(this)

            updateCalendarIdSpinner()

        }

    }

    private fun updateCalendarIdSpinner() {
        if (calendarIdTable == null) {
            return
        }

        val list = ArrayList<String>()

        val e = calendarIdTable!!.keys()
        while (e.hasMoreElements()) {
            val key = e.nextElement() as String
            list.add(key)
        }

        val dataAdapter = ArrayAdapter(this,
                android.R.layout.simple_spinner_item, list)
        dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        calendarIdSpinner!!.adapter = dataAdapter
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {

        if (requestCode == CalendarHelper.CALENDARHELPER_PERMISSION_REQUEST_CODE) {
            if (CalendarHelper.haveCalendarReadWritePermissions(this)) {
                Toast.makeText(this, "Have Calendar Read/Write Permission.",
                        Toast.LENGTH_LONG).show()
                updateCalendarIdSpinner()
            }

        }

        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }

    private fun addNewEvent() {
        if (calendarIdTable == null) {
            Toast.makeText(this, "No calendars found. Please ensure at least one google account has been added.",
                    Toast.LENGTH_LONG).show()
            //Load calendars
            calendarIdTable = CalendarHelper.listCalendarId(this)

            updateCalendarIdSpinner()

            return
        }

        val oneHour = (1000 * 60 * 60).toLong()
        val tenMinutes = (1000 * 60 * 5).toLong()

        val oneHourFromNow = Date().time + oneHour
        val tenMinutesFromNow = Date().time + tenMinutes

        CalendarHelper.MakeNewCalendarEntry(6589, this, "Test", "Add event", "Somewhere", tenMinutesFromNow, tenMinutesFromNow + oneHour, false, true, CalendarHelper.getCalendarId(this), 3)

    }

    companion object {

        private val TAG = "MainActivity"
    }


}
